section4/reflection.md

```diff
@@ -1,21 +1,39 @@
 
-1. What is an Object, and how is it different from an Array in Javascript?
+3. What is an Object, and how is it different from an Array in Javascript?
+  * An Object is like a database for storing and organizing data of any data type. An Object and an Array are very similar. They are both mutable and can have different actions performed on them to manipulate their data. It seems that the biggest difference is that an Array is more like an itemized list, while an Object is a more a "thing" with defined (changable) properties for the data.
 
-1. For each set of data, would an array or object be better to store it? Explain your choice.
+4. For each set of data, would an array or object be better to store it? Explain your choice.
 
   * List of all of the students in class
+    - Array: just a list, no properties along with them
   * List of states and their capitals
+    - Object: could easily be set in a key:value pairing for nice organization
   * List of things to pack for vacation
+    - Array: itemized list.
   * Names of all the Instagram accounts I follow
+    - Array: itemized list.
   * List of student names and their cohort
+    - Object: key:value pairing with student: cohort for organization
   * Ingredients and amount of each ingredient to bake a cake
+    - Object: key being the ingredients and the amount of each being the value.
   * All my favorite restaurants
+    - Array: itemized list of restaurants.
 
-1. In this section, we talked about an `transaction item object`. It has keys that represent different properties an bank transaction would have, and values that store the data. In a banking app, this data would eventually be displayed in the browser for the user. Think of another Object in a web application that you use frequently. What is it? What keys might be on it? What are example values? 
+5. In this section, we talked about a `transaction item object`. It has keys that represent different properties a bank transaction would have, and values that store the data. In a banking app, this data would eventually be displayed in the browser for the user. Think of another Object in a web application that you use frequently. What is it? What keys might be on it? What are example values?
 
-1. What questions do you still have about classes and/or Objects?
+  * Any trip planning website would have this. Sadly, I don't currently use those frequently at the moment. Let's say Expedia. Key and value examples would be:
+    - departureDate : '4/4/21',
+    - returnDate: '4/11/21',
+    - numberOfPassengers: 3,
+    - carRental: false,
+    - hotelRoom: true,
+
+6. What questions do you still have about classes and/or Objects?
+  * Is there a particular reason that classes use PascalCase?
```

Comment:
Since JS is case sensitive it helps with how the machine reads the code. It says, "Oh look, a capital letter is beginning this phrase... hmmm... It must be a class!" and then assigns the behavior of a class. VS when it comes across a lower case letter, it begins figuring out if it is a variable or a function, etc.